salesagility/SuiteCRM
# Add Encrypt fields to Module Builder

## Description







In Module Builder, add support for Encrypt field which was removed in SugarCRM CE 6.1 in 2011.

Solves issue #1755 

See forum request:
https://www.suitecrm.com/forum/developer-help/1740-securing-encrypting-a-custom-field
## Motivation and Context



These encrypted fields are required by law in many countries, for storing identity information, such as, social security number, drivers license number, date of birth, credit card number, etc.

SuiteCRM needs to comply with national laws requiring encrypted fields to safely contain personally identifying information, which is highly sought after by hackers and other unscrupulous sysadmins with root access seeking to commit financial fraud, identity theft, etc, by dumping the database and stealing such information needed to purchase items in the name of someone else.
## How To Test This



Go into Module Builder.  
You should be able to select field of type "Encrypt", label it "Social Security Number".
Add the field to your Quick Create, Detail and Edit views.
Save and click Deploy to make the the module active in your site.
Go into the module.
Create a new record.
Enter data into the Social Security Number field.
Save the new record.
Browse the database with phpMyAdmin.
Find the field for the new record you created.
The column for Social Security Number should be an illegible string of characters.
## Types of changes


### Final checklist
